src/ripple/protocol/AmountConversions.h

```diff
@@ -130,16 +132,48 @@ toAmount(
     saveNumberRoundMode rm(Number::getround());
     if (isXRP(issue))
         Number::setround(mode);
+
     if constexpr (std::is_same_v<IOUAmount, T>)
         return IOUAmount(n);
-    if constexpr (std::is_same_v<XRPAmount, T>)
+    else if constexpr (std::is_same_v<XRPAmount, T>)
         return XRPAmount(static_cast<std::int64_t>(n));
-    if constexpr (std::is_same_v<STAmount, T>)
+    else if constexpr (std::is_same_v<STAmount, T>)
     {
         if (isXRP(issue))
             return STAmount(issue, static_cast<std::int64_t>(n));
         return STAmount(issue, n.mantissa(), n.exponent());
     }
+    else
+    {
+        constexpr bool alwaysFalse = !std::is_same_v<T, T>;
+        static_assert(alwaysFalse, "Unsupported type for toAmount");
+    }
+}
+
```

Comment:
How about using `concept` instead of the `static_assert` here and other places?

    template <typename T>
    concept ValidAmount =
        std::is_same_v<T, IOUAmount> ||
        std::is_same_v<T, XRPAmount> ||
        std::is_same_v<T, STAmount>;

    template <ValidAmount T>
    T
    toMaxAmount(Issue const& issue)
    {
     ...
    }

I don't think max `Number` is applicable, at least not in the context that `toMaxAmount()` is used.

Reply:
I thought I got a compile error without the supporting `Number`.

I don't object to the `concept`, but I weakly prefer the code without it. I don't love that the `is_same` is repeated in both the body of the function and the concept, and the concept isn't important for distinguishing between overloads. I'll change it if you or others feel strongly, but I think I prefer the non-concept version.

Reply:
It builds for me without `Number` on OSX, Linux, and Windows.

I prefer `concept` since it's clearly defines the type constraint. I don't feel strongly about it though.

Reply:
I hear you. I don't think this is worth spending too much time on. I'm going to keep it as is.